src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -70,6 +68,7 @@ namespace OpenMS
     public RangeManager<1>,
     public SpectrumSettings
   {
+     friend class MSSpectrumHelper;
```

Comment:
hmm design doesn't feel right to me. You have public read access to the dataarrays (e.g. getFloatDataArrays () const and mutable).